devops/jobs/ImageBuilder.groovy

```diff
@@ -0,0 +1,107 @@
+/*
+    This job serves to build an application's Docker image and to push it to Docker Hub. This job only serves as a downstream job to the 
+    various <app>-watcher jobs and does not run on its own. This job receives the name of an application as a build parameter. 
+    This application's Docker image is then built and pushed.
+
+    Upon a push to master, the job is triggered via a webhook, at which point it runs the parsefiles.py script with the files changed between 
+    the old commit and the new commit. This resolves which Docker images need to be rebuilt as a result of the aforementioned changes to the 
+    configuration repository. For each Docker image that needs rebuilding, this job triggers the respective <app>-watcher job.
+
+    Variables consumed from the EXTRA_VARS input to your seed job in addition
+    to those listed in the seed job.
+    APPS_TO_CONFIG: a dictionary containing mappings from an edX IDA to a dictionary of various configuration values (REQUIRED). 
+        The structure of APPS_TO_CONFIG should be
+        
+        APPS_TO_CONFIG:
```

Comment:
I think some of these docs are also more for AppWatcher, at least from reading the code